### PR TITLE
Support wrangler2 - v2

### PR DESCRIPTION
**Pull request by WalshyDev**
Whooo! Wrangler 2 is awesome :)

Since this is v2 I did make 2 breaking changes but the migration path is super simple.
* Remove `publish: true` with `command: publish`.

---

You can see it working for a Worker publish here: https://github.com/WalshyDev/wrangler-action-test/runs/6310845616?check_suite_focus=true

You can see it working for a Pages deployment here: https://github.com/WalshyDev/wrangler-action-test/runs/6327883564?check_suite_focus=true

cc: @GregBrimble @eidam @Ekwuno

### CODE REVIEW

File: action.yml
Comment (GregBrimble) [inline action.yml:41]: I think we probably want the default `command` to be `publish`, just so it's as little friction for people upgrading from v1 as possible. I think wrangler2 will still grab the entrypoint from the various places it can be specified (although you might get warnings to suggest you instead make it explicit in the command).

File: entrypoint.sh
Comment (GregBrimble) [inline entrypoint.sh:42]: I'd flip this around. So:

- install v1 if `INPUT_WRANGLERVERSION == 1*`
- else, use v2.

We might introduce tags other than `beta` e.g. `wasm` or `pages` as we prototype stuff.

File: action.yml
Comment (eidam) [reply action.yml:41]: Agree, I would be for having `command` optional, and you are already setting it as default when the input is not present :)
https://github.com/cloudflare/wrangler-action/pull/72/files#diff-6f9d41d046756f0ddc2fcee0626bdb50100d12b88f293734eff742818e03efa2R126

File: action.yml
Comment (eidam) [inline action.yml:29]: should this be `2.0.0`? 

```
npm ERR! code ETARGET
npm ERR! notarget No matching version found for wrangler@2.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.
```

Comment (eidam) [reply action.yml:29]: nvm, we dont have 2.x.x yet

File: action.yml
Comment (eidam) [inline action.yml:16]: I think we need to keep `environment` and pass it to all commands by default (like `wrangler secret put`). I just tried to deploy my Workers with `wrangler publish --env production`, but all secrets were created for the default environment (without `--production` suffix)

File: README.md
Comment (deadlypants1973) [inline README.md:163]:
```suggestion
Note that there are a number of possible events, like `push`, that can be used to trigger a workflow. For more details on the events available, refer to the [GitHub Actions documentation](https://help.github.com/en/articles/workflow-syntax-for-github-actions#on).
```

File: README.md
Comment (deadlypants1973) [inline README.md:167]:
```suggestion
If you want to deploy your Pages project with GitHub Actions rather than the built-in CI, then this is a great way to do it. Wrangler v2 will populate the commit message and branch for you. You only need to pass the project name. If a push to a non-production branch is done, it will publish as a preview deployment:
```

File: README.md
Comment (deadlypants1973) [inline README.md:188]:
```suggestion
If you would like to deploy your Workers application on a recurring basis – for example, every hour, or daily – the `schedule` trigger allows you to use cron syntax to define a workflow schedule. The below example will deploy at the beginning of every hour:
```

File: README.md
Comment (deadlypants1973) [inline README.md:210]:
```suggestion
### Deploying on a dispatched event
```

File: README.md
Comment (deadlypants1973) [inline README.md:212]:
```suggestion
If you need to trigger a workflow at-will, you can use GitHub's `workflow_dispatch` [event](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#workflow_dispatch) in your workflow file. By setting your workflow to trigger on that event, you will be able to deploy your application via the GitHub UI. The UI also accepts inputs that can be used to configure the action:
```

File: README.md
Comment (deadlypants1973) [inline README.md:235]:
```suggestion
For more advanced usage or to programmatically trigger the workflow from scripts, refer to [the GitHub documentation](https://docs.github.com/en/rest/reference/actions#create-a-workflow-dispatch-event) for making API calls.
```

File: README.md
Comment (deadlypants1973) [inline README.md:241]:
```suggestion
Refer to the [Quick Start guide](https://developers.cloudflare.com/workers/quickstart) to get started. Once you have a Workers application, you may want to set it up to automatically deploy from GitHub whenever you change your project.
```

File: README.md
Comment (deadlypants1973) [inline README.md:245]:
```suggestion
You will need to add `account_id = ""` in your `wrangler.toml` file or set `accountId` in this GitHub Action.
```

File: action.yml
Comment (deadlypants1973) [inline action.yml:5]:
```suggestion
description: "Deploy your Cloudflare Workers and Pages projects from GitHub using Wrangler"
```

File: CHANGELOG.md
Comment (deadlypants1973) [inline CHANGELOG.md:12]:
```suggestion
## Removals

```

File: CHANGELOG.md
Comment (deadlypants1973) [inline CHANGELOG.md:16]:
```suggestion
## Changes

```

**Review by deadlypants1973 — COMMENTED** [review]
PCX review

File: README.md
Comment (WalshyDev) [reply README.md:9]: On a git push, how's this?
```suggestion
Add `wrangler-action` to the workflow for your Workers/Pages application. The below example will publish a Worker on a git push to the `main` branch:
```

File: entrypoint.sh
Comment (petebacondarwin) [inline entrypoint.sh:49]: Should this be 2?

```suggestion
  WRANGLER_VERSION=2
```

File: entrypoint.sh
Comment (petebacondarwin) [inline entrypoint.sh:55]: Wrangler 2 still supports the `CF_...` environment vars for the time being. 
So you could avoid these checks if you like.

File: entrypoint.sh
Comment (petebacondarwin) [inline entrypoint.sh:74]: What uses these env vars? Not Wrangler AFAIK.

File: action.yml
Comment (petebacondarwin) [inline action.yml:39]: Perhaps highlight that this will be a string?

```suggestion
    description: "The Wrangler command you wish to run. For example: `\"publish\"` - this will publish your Worker"
```

File: action.yml
Comment (petebacondarwin) [reply action.yml:16]: That sounds right. I agree

File: README.md
Comment (petebacondarwin) [reply README.md:210]: I think that mentioning dispatch event here at all is an implementation concern.
Instead what the user wants to do here is trigger a deploy manually...

```suggestion
### Manually triggering a deployment
```

File: README.md
Comment (petebacondarwin) [inline README.md:167]: What decides if a branch is "production" or not?

File: README.md
Comment (petebacondarwin) [inline README.md:126]: `wrangler build` is no longer a thing in Wrangler 2.
Perhaps remove this sentence?

**Review by petebacondarwin — CHANGES_REQUESTED** [review]
Nice update @WalshyDev - I think the action should still support setting the environment.

File: entrypoint.sh
Comment (WalshyDev) [reply entrypoint.sh:55]: Yeah, I just wanted to avoid any possible future breakage and the deprecation warnings. Not a big deal doing it this way

File: action.yml
Comment (WalshyDev) [reply action.yml:39]: good call

File: README.md
Comment (WalshyDev) [reply README.md:167]: If they create the project through CLI, they can configure it. The dashboard defaults to using "production".

File: README.md
Comment (WalshyDev) [reply README.md:126]: I think it's still good to have this little bit, should just be a different command. Not sure what that command should be right now though...

File: entrypoint.sh
Comment (WalshyDev) [inline entrypoint.sh:143]: Open question: I don't think it's feasible for me to pass --env here as the user may also do that or it may result in different behaviour than intended (or even an unknown option warning/error).

How do we handle the env here?

File: README.md
Comment (GregBrimble) [inline README.md:205]: I'd get rid of all of these from the docs to be ✨ clean ✨ if they're not required now :)

**Review by GregBrimble — APPROVED** [review]
LGTM. Left that one suggestion for the docs, but not-blocking, imo.

I'd wait for one or two more people to approve, and then we can merge!

**Comment by WalshyDev** [comment]
By the way, I don't have write access on this repo so I cannot merge or cut a release.

@codewithkristian or @eidam would probably need to

File: entrypoint.sh
Comment (petebacondarwin) [reply entrypoint.sh:143]: Perhaps throw an error if both INPUT_COMMAND and INPUT_ENVIRONMENT are set, saying that if you are setting your own command then you must pass the environment as a command line arg?

**Review by petebacondarwin — APPROVED** [review]

File: CHANGELOG.md
Comment (deadlypants1973) [inline CHANGELOG.md:14]:
```suggestion
* Removed `publish` input (refer to [Breaking changes](#breaking-changes)).
```

File: README.md
Comment (deadlypants1973) [inline README.md:9]:
```suggestion
Add `wrangler-action` to the workflow for your Workers/Pages application. The below example will publish a Worker on a `git push` to the `main` branch:
```

File: README.md
Comment (deadlypants1973) [inline README.md:165]:
```suggestion
### Deploy your Pages site (production & preview)
```

File: README.md
Comment (deadlypants1973) [inline README.md:140]:
```suggestion
### Deploy when commits are merged to main
```

File: README.md
Comment (deadlypants1973) [inline README.md:167]:
```suggestion
If you want to deploy your Pages project with GitHub Actions rather than the built-in CI, then this is a great way to do it. Wrangler 2 will populate the commit message and branch for you. You only need to pass the project name. If a push to a non-production branch is done, it will publish as a preview deployment:
```

What is a CI? Need to add full noun then (CI) in this first sentence.

File: entrypoint.sh
Comment (deadlypants1973) [inline entrypoint.sh:46]:
```suggestion
# Else install Wrangler 2
```

File: README.md
Comment (WalshyDev) [reply README.md:109]:
```suggestion
If you need to run additional shell commands before or after your command, you can specify them as input to `preCommands` (before `publish`) or `postCommands` (after `publish`). These can include additional `wrangler` commands (that is, `whoami`, `kv:key put`) or any other commands available inside the `wrangler-action` context.
```

File: entrypoint.sh
Comment (WalshyDev) [reply entrypoint.sh:147]: I can't tell if the command has --env but we still need INPUT_ENVIRONMENT for secrets. Erroring on both existing wouldn't work since that's ok, I can add a notice but it may be a little annoying for those who do have it setup correctly. Will add a notice for now

**Review by deadlypants1973 — APPROVED** [review]

**Comment by eidam** [comment]
I can merge it in 30

**Pull request merged**
